feat(distro): add StackdriverJSONLayout appender

menski opened the pull request:
## Description

To enable stackdriver compatible JSON logging set env variable `ZEEBE_LOG_APPENDER` to `Stackdriver`.

Remove log4j2 configuration location from startup parameters to allow overriding with custom file locations.

Enable stackdriver logging in benchmarks per default to improve log discovery.

Add documentation how to enable stackdriver logging.

## Related issues

closes #4190 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

pihme commented on docs/src/operations/configuration.md:
````diff
@@ -114,3 +114,18 @@ Action:
 Update your application's configuration
 ```
 
+# Logging
````
Optional comments:
- not sure about "Logging" as top level header, I would probably move it one level down
- The last section on Google Stackdriver gets lost; would move it above 'Default Logging Configuration'

pihme commented on util/src/test/java/io/zeebe/util/StackdriverJSONLayoutTest.java:
```diff
@@ -0,0 +1,71 @@
+    assertThat(jsonMap).containsEntry("logger", logger.getName());
+  }
+
+  private Map<String, String> logOutputToJSONMap() throws IOException {
```
Test is difficult to read. 
I would prefer if both `logOutputToJSONMap()` and `createTestAppenderWithlayout()` both have a writer as parameter. 
Then it would be clearer what the test is doing. 

Maybe make logOutput a local variable in the test.

pihme commented on util/src/test/java/io/zeebe/util/StackdriverJSONLayoutTest.java:
```diff
@@ -0,0 +1,71 @@
+        .readValue(logOutput.toString());
+  }
+
+  private void createTestAppenderWithLayout(StringLayout layout) {
```
rename to `register/startTestAppender...` or return the appender

pihme commented on util/src/test/java/io/zeebe/util/StackdriverJSONLayoutTest.java:
```diff
@@ -0,0 +1,71 @@
+    final Appender appender =
+        WriterAppender.createAppender(layout, null, logOutput, "test", false, false);
+    final LoggerConfig loggerConfig = context.getConfiguration().getRootLogger();
+    loggerConfig.addAppender(appender, Level.ALL, null);
```
Missing `loggerConfig.removeAppender(...)` as part of test cleanup. Or use a logger local to this test

pihme commented on util/src/test/java/io/zeebe/util/StackdriverJSONLayoutTest.java:
```diff
@@ -0,0 +1,71 @@
+    logger.error("Should appear as JSON formatted ouput");
+    // then
+    final Map<String, String> jsonMap = logOutputToJSONMap();
+    assertThat(jsonMap)
```
Optional: use soft asserts

menski replied:
Thanks I learned something new :sparkles:

menski commented on util/src/main/java/io/zeebe/util/StackdriverJSONLayout.java:
```diff
@@ -0,0 +1,87 @@
+@Plugin(name = "StackdriverJSONLayout", category = Node.CATEGORY, elementType = Layout.ELEMENT_TYPE)
+public class StackdriverJSONLayout extends AbstractStringLayout {
+
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
```
I would not mix this, as the `ObjectWriterFactory` is intend to be configured to print a specific format. The Logger format is simple and is independent from what we are needing somewhere else in the code base. We also are not transforming date objects in the logger format, just longs and strings.

menski commented:
Thanks Peter for the fast review. I addressed your comments in multiple new commits for you to review. I plan to squash them before merge, so give me a heads-up if there is anything else I should do.

pihme commented on docs/src/appendix/environment-variables.md:
```diff
@@ -1,19 +1,23 @@
-  - `ZEEBE_LOG_LEVEL`: Sets the log level of the Zeebe Logger (default: `info`).
- 
- ## Environment Variables for Developers 
+## Environment Variables for Operators The following environment variables are
```
missing line break

menski commented:
bors r+
